lib/facter/self_service.rb

```diff
@@ -236,6 +236,13 @@
     { S0033: hiera_config_file.dig('version') == 5 }
   end
 
+  chunk(:S0035) do
+    # restrict to primary/replica/compiler
+    next unless PuppetSelfService.replica? || PuppetSelfService.compiler? || PuppetSelfService.legacy_compiler? || PuppetSelfService.primary?
+    # return false if any Warnings appear in the 'puppet module list...'
```

Comment:
I wonder if it would be better to use the `puppet module` face for this than shelling out.  These are callable from Ruby code like this: 

```
require 'puppet'
require 'puppet/face'

face = Puppet::Face[:module, :current]
modules = face.list(environment: 'production')

# Each key is the modulepath, and its value is an array of the modules
has_warnings = modules[:modules_by_path].any? {|k,v|
  v.any? {|mod|
    !mod.unmet_dependencies.empty?
  }
}
```
I tested this in a couple of scenarios and it seems to work.  It's also more accurate, since it's possible for a user to have a module with the word "Warning" in it, which I know is unlikely, but still possible.